batch-register3.py

```diff
@@ -214,335 +226,420 @@
 # library for better portability.
 import xml.etree.ElementTree
 
-def loadMappings (file):
-  # returns: [(destination, expression), ...]
-  m = []
-  with open(file) as f:
-    n = 0
-    for l in f:
-      n += 1
-      try:
-        assert "=" in l, "invalid syntax"
-        d, e = [v.strip() for v in l.split("=", 1)]
-        if d.startswith("/"):
-          assert re.match("/resource(/\w+)+(@\w+)?$", d),\
-            "invalid absolute XPath expression"
+import argparse
+import base64
+import csv
```

Comment:
Repeating with line 216-218

Reply:
Thank you for chatching this Jing, I'll clean it up.